pydantic/_internal/_generate_schema.py

```diff
@@ -1338,12 +1342,13 @@ def _type_alias_type_schema(self, obj: TypeAliasType) -> CoreSchema:
                 return maybe_schema
 
             origin: TypeAliasType = get_origin(obj) or obj
-
-            annotation = origin.__value__
```

Comment:
Moved below, as accessing the `__value__` attribute will trigger the evaluation of the annotation, and thus this can raise a `NameError`. The `eval_type` call is still necessary though, as you can have explicit string references inside a a type alias type.

Reply:
Maybe I'm misunderstanding - was this moved below? Or do we not need this specific line anymore?

Reply:
Basically, here is what can happen with type aliases:

```python
type A = Int

annotation = origin.__value__
# NameError, 'Int' is not defined

type B = 'Int'
annotation = origin.__value__
assert annotation == 'Int'

eval_type(B, ..., ...)
# NameError, 'Int' is not defined
```

So everything was moved in a single `try..except` block:

```python
try:
    annotation = _typing_extra.eval_type(origin.__value__, *self._types_namespace)
except ...: ...
```
```